### ISSUE TITLE
[FEATURE] Point wash at locally compiled version of wasmcloud

### ISSUE THREAD

**Issue by cdmurph32** [wasmCloud/wasmCloud#3089]
### Affected project(s)

- [ ] documentation
- [ ] examples
- [ ] wasmCloud host
- [X] wasmCloud CLI (wash)
- [ ] wasmCloud dashboard UI (washboard)
- [ ] capability providers
- [ ] provider bindgen
- [ ] control interface client
- [ ] other / not sure

### Is your feature request related to a problem? Please describe.

When I am working on wasmcloud, it would be useful to start the locally compiled version with wash. This could also be useful for customer support or debugging, as wasmcloud binaries could be used directly.

### Describe the solution you'd like

`--wasmcloud-path` argument which is given a path to the wasmcloud binary

### Describe alternatives you've considered

No new feature: Use the workaround of placing the local wasmcloud version in ~/.wash/downloads with a custom path.
No new feature: Run `wash up`, start the local wasmcloud, and then stop the wasmcloud host that wash started.
Add `--no-host` argument to wash so that wash does not start a wasmcloud host.

### Additional context

https://wasmcloud.slack.com/archives/CS38R7N9Y/p1726847907598719

**Comment by brooksmtownsend** [comment]
I'm a fan of this. It's often useful for wash to be able to launch wadm + NATS for local development, and while `--no-host` would be one way to accomplish this, it's a little awkward to have wash start these two services without a host.

So the user experience here @cdmurph32 would be something like (using my file setup as an example):
```bash
wash up --host-path ~/github.com/wasmcloud/wasmcloud/target/debug/wasmcloud
```

**Comment by lachieh** [comment]
I like `--host-path`. Do we also want to add an env variable? `WASMCLOUD_HOST_PATH` or something?

**Comment by vados-cosmonic** [comment]
Hey @cdmurph32 just to let you know this was implemented! Going to get it released in the next version (likely 0.34.1) of `wash`
